### ISSUE TITLE
ffmpeg problem

### ISSUE THREAD

**Issue by DrDreGFunk** [deepfakes/faceswap-playground#83]
I've searched for the code to put the images into a video but it's not working. The code I am using is:

ffmpeg -i video%04d.png -c:v libx264 -vf "fps=30,format=yuv420p" video.mp4

All the images in the folder are named: video1.png ... and so on. 

This is the error I get back in the terminal:

------------------------------------------------------------------------------------------------------
MY NAME-MacBook-Pro-2:faceswap-master MY NAME$ ffmpeg -i video%04d.png -c:v libx264 -vf "fps=30,format=yuv420p" video.mp4

ffmpeg version 3.4.1 Copyright (c) 2000-2017 the FFmpeg developers

  built with Apple LLVM version 9.0.0 (clang-900.0.39.2)

  configuration: --prefix=/usr/local/Cellar/ffmpeg/3.4.1 --enable-shared --enable-pthreads --enable-version3 --enable-hardcoded-tables --enable-avresample --cc=clang --host-cflags= --host-ldflags= --enable-gpl --enable-libmp3lame --enable-libx264 --enable-libxvid --enable-opencl --enable-videotoolbox --disable-lzma

  libavutil      55. 78.100 / 55. 78.100

  libavcodec     57.107.100 / 57.107.100

  libavformat    57. 83.100 / 57. 83.100

  libavdevice    57. 10.100 / 57. 10.100

  libavfilter     6.107.100 /  6.107.100

  libavresample   3.  7.  0 /  3.  7.  0

  libswscale      4.  8.100 /  4.  8.100

  libswresample   2.  9.100 /  2.  9.100

  libpostproc    54.  7.100 / 54.  7.100

[image2 @ 0x7f7f63800000] Could find no file with path ‘video%04d.png' and index in the range 0-4

video%04d.png: No such file or directory

------------------------------------------------------------------------------------------------------

What should I use instead?
Also, some of my images are not all converted with my Data_B face. Is that normal?

Thanks in advance!

**Comment by MT-4ier** [comment]
The %04d is the number of digits it's looking for in the first filename, so it's trying to find video0001.png.
It's recommended to have leading 0s on your filenames because ffmpeg will put frames 10-19 after frame 1, then frame 2, 20-29, frame 3, and so on.
Rename your files to have enough leading 0s so that the highest number and the first number have the same number of digits, then alter the 4 in %04d.png to match that number of digits.

**Comment by DrDreGFunk** [comment]
Ok thank you! Is there a quick way to rename these files as there are 930 of them. Also, some of my images are not all converted with my Data_B face. Is that normal?

**Comment by MT-4ier** [comment]
I use Bulk Rename Utility, but it looks like it's Windows only. You can search for a Mac bulk renaming program.
For the images that weren't converted, were the faces extracted from them in the first place? If there was no face detected, or if you were filtering for a specific face and it didn't detect that face in that image, it won't have made a file for training and re-merging.
That's my best guess as to what happened.

**Comment by DrDreGFunk** [comment]
@MT-4ier Yh the faces were all extracted. There were even some frames where the face was very obvious to detect. And yh I still got that same error (I don't know if you read the previous comment as I just edited it).

**Comment by MT-4ier** [comment]
Try
ffmpeg -f image2 -i video%03d.png -c:v libx264 -vf "fps=30,format=yuv420p" video.mp4

Another thing to check is if the command prompt / terminal is at the correct path/directory where your images are located.

**Comment by DrDreGFunk** [comment]
IT WORKED! THANK YOU SO MUCH BRO!

**Comment by MT-4ier** [comment]
That's great!
I think the "-f image2" told ffmpeg that it was looking for images, and that's why it worked.
:)

**Comment by iperov** [comment]
my windows script to detect fps from video

```
for /f "tokens=*" %%a in (
    '_internal\bin\ffmpeg\ffmpeg.exe -i workspace\data_dst.mp4 2^>^&1 ^| find "fps"'
) do (
    for /f "tokens=6 delims=," %%0 in (
        "%%a"
    ) do (
        SET x=%%0
        SET fps=!x:~1,-3%!
    )    
)
```

then use it in build avi 

`_internal\bin\ffmpeg\ffmpeg.exe -r !fps! -i %WORKSPACE%\data_dst\merged\out%%5d.png -r !fps! -c:v libx264 -crf 16 %WORKSPACE%\result.avi`

**Comment by DrDreGFunk** [comment]
Closed #83
